### DISCUSSION TITLE
Generate Code - Seems to need Access to Resources

### DISCUSSION THREAD

**Discussion by chrismalek** [AnWeber/vscode-httpyac#67] [Q&A] [answered]
Thanks again for the great vscode extension. I use it every day.

I often need to generate Curl commands from my HTTP files to give to other people to run. On my workstation, I often do not have access to the endpoints due to security restrictions.

It seems when I try the "Generate Code" command HTTPYAC seems to need to actually execute the command to generate the curl command. Is there a way to have it resolved all the variables and generate the curl without being able to connect to the actual end points?

**Comment by AnWeber** [comment, marked as answer]
Too bad. The place I have changed exactly, in the last version, that I generate code while logging the response. It also goes before, only it was.so easier and I thought it makes no problem. I will change it again and adjust the timing.

**Reply by chrismalek** [reply]
Thanks I will wait for the next release.I am a watcher on the repository.

**Reply by AnWeber** [reply]
The behavior was also reported to me as part of Issue #68. I have provided an update with a fix. 

Since the change was noticed so quickly, apparently the feature is more important to you. I added a few more improvements to code generation:

* new Setting httpyac.generateCodeDefaultLanguage to set default Language for Code Generation
* new Setting httpyac.generateCodeTargetOutput to specify Output Target of generated code (Clipboard or Window)
* new Command httpyac.generateCodeSelectLanguage to always select Language of Code Generation
* CodeLens support for httpyac.generateCode and httpyac.generateCodeSelectLanguage

**Reply by chrismalek** [reply]
Cool I will check it out!!!   Love it.

**Reply by st11x** [reply]
@AnWeber I'm using v 5.7.2. 

The first generate code request in whichever language works. It brings up a new tab with the code. If you then repeat the command, it does not work anymore. Neither does generating code in any other languages. 

It gets "fixed" if you close the .http file and reopen it, but again, it only works for the first request. 

I don't see any errors in the console. Do you need the CLI installed?

**Reply by AnWeber** [reply]
@st11x I could reproduce it. I'll fix it with the next version. (see https://github.com/AnWeber/hookpoint/issues/3)
